env0/resource_configuration_variable.go

```diff
@@ -141,27 +141,36 @@ func whichScope(d *schema.ResourceData) (client.Scope, string) {
 	return scope, scopeId
 }
 
-func resourceConfigurationVariableCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+func getConfigurationVariableCreateParams(d *schema.ResourceData) (*client.ConfigurationVariableCreateParams, error) {
 	scope, scopeId := whichScope(d)
 	params := client.ConfigurationVariableCreateParams{Scope: scope, ScopeId: scopeId}
 	if err := readResourceData(&params, d); err != nil {
-		return diag.Errorf("schema resource data deserialization failed: %v", err)
+		return nil, fmt.Errorf("schema resource data deserialization failed: %v", err)
 	}
 
 	if err := validateNilValue(params.IsReadOnly, params.IsRequired, params.Value); err != nil {
-		return diag.Errorf(err.Error())
+		return nil, err
 	}
 
-	actualEnumValues, getEnumErr := getEnum(d, params.Value)
-	if getEnumErr != nil {
-		return getEnumErr
+	actualEnumValues, err := getEnum(d, params.Value)
+	if err != nil {
+		return nil, err
 	}
 
 	params.EnumValues = actualEnumValues
 
+	return &params, nil
+}
+
+func resourceConfigurationVariableCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
```

Comment:
The git diff is a little confusing.
Basically took some of the functionality in here and moved it to the common function  getConfigurationVariableCreateParam

Now both update and create use it.